components/board/pi/impl/board.go

```diff
@@ -101,110 +120,99 @@ func NewPigpio(ctx context.Context, name resource.Name, cfg *genericlinux.Config
 		return nil, picommon.ConvertErrorCodeToMessage(int(resCode), "gpioCfgSetInternals failed with code")
 	}
 
-	// setup
-	cancelCtx, cancelFunc := context.WithCancel(ctx)
+	if err := initializePigpio(); err != nil {
+		return nil, err
+	}
+
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	piInstance := &piPigpio{
 		Named:           name.AsNamed(),
-		cfg:             cfg,
 		logger:          logger,
 		isClosed:        false,
 		interruptCtx:    cancelCtx,
 		interruptCancel: cancelFunc,
 	}
 
-	instanceMu.Lock()
-
-	// if pigpio is not initialized, only then we initialize it.
-	if !pigpioInitialized {
-		resCode = C.gpioInitialise()
-		if resCode < 0 {
-			instanceMu.Unlock()
-			// failed to init, check for common causes
-			_, err := os.Stat("/sys/bus/platform/drivers/raspberrypi-firmware")
-			if err != nil {
-				return nil, errors.New("not running on a pi")
-			}
-			if os.Getuid() != 0 {
-				return nil, errors.New("not running as root, try sudo")
-			}
-			return nil, picommon.ConvertErrorCodeToMessage(int(resCode), "error")
-		}
+	if err := piInstance.performConfiguration(ctx, nil, cfg); err != nil {
+		C.gpioTerminate()
+		logger.Debug("Pi GPIO terminated due to failed init.")
```

Comment:
Error here please.

Reply:
Agreed! The original code was a debug log, but you're right it should be changed. (reference back to your previous question: this is what happened to the code that was originally related to `initGood`).